lib/scrivener/paginater/ecto/query.ex

```diff
@@ -26,25 +28,59 @@ defimpl Scrivener.Paginater, for: Ecto.Query do
     page_number =
       if allow_overflow_page_number, do: page_number, else: min(total_pages, page_number)
 
+    entries =
+      if page_number > total_pages,
+        do: [],
+        else: entries(query, repo, page_number, page_size, options)
+
     %Page{
       page_size: page_size,
       page_number: page_number,
-      entries: entries(query, repo, page_number, total_pages, page_size, options),
+      entries: entries,
       total_entries: total_entries,
       total_pages: total_pages
     }
   end
 
-  defp entries(_query, _repo, page_number, total_pages, _page_size, _options)
-       when page_number > total_pages,
-       do: []
+  def paginate(query, %Config{
+        page_type: :simple,
+        module: repo,
+        page_number: page_number,
+        page_size: page_size,
+        options: options
+      }) do
+    entries_with_maybe_one_extra =
+      entries(query, repo, page_number, page_size, options, extra_entry_size: 1)
```

Comment:
Here is the core logic where we fetch one more row than it should be.

If the total fetched entries are > page_size, we set has_more to true(and discard the last one from entries). Other, has_more is false